SDKAndroid/src/main/java/app/cybrid/sdkandroid/components/BankAccountsView.kt

```diff
@@ -0,0 +1,442 @@
+
+    var bankAccountViewModel: BankAccountsViewModel? = null
+
+    init {
+
+        LayoutInflater.from(context).inflate(R.layout.bankaccounts_component, this, true)
+        this.composeView = findViewById(R.id.composeContent)
+    }
+
+    fun setViewModel(bankAccountViewModel: BankAccountsViewModel) {
+
+        this.bankAccountViewModel = bankAccountViewModel
+        this.currentState = bankAccountViewModel.uiState
+        this.initComposeView()
+        GlobalScope.launch { bankAccountViewModel.createWorkflow() }
+    }
+
+    private fun initComposeView() {
+
+        this.composeView?.let { compose ->
+            compose.setContent {
+                BankAccountsView(
+                    currentState = this.currentState,
+                    viewModel = bankAccountViewModel
+                )
+            }
+        }
+    }
+
+    companion object {
+
+        fun openPlaid(plaidToken: String, getPlaidResult: ManagedActivityResultLauncher<LinkTokenConfiguration, LinkResult>) {
+
+            val linkTokenConfiguration = linkTokenConfiguration {
+                token = plaidToken
+            }
+            getPlaidResult.launch(linkTokenConfiguration)
+        }
+    }
+}
+
+/**
+ * Composable Function for Bank Accounts
+ **/
+
+@Composable
+fun BankAccountsView(
+    currentState: MutableState<BankAccountsView.BankAccountsViewState>,
+    viewModel: BankAccountsViewModel?) {
+
+    // -- Content
+    Surface {
+
+        when(currentState.value) {
+
+            BankAccountsView.BankAccountsViewState.LOADING -> {
+                BankAccountsView_Loading()
+            }
+
+            BankAccountsView.BankAccountsViewState.REQUIRED -> {
+                BankAccountsView_Required(viewModel)
+            }
+
+            BankAccountsView.BankAccountsViewState.DONE -> {
+                BankAccountsView_Done()
+            }
+
+            BankAccountsView.BankAccountsViewState.ERROR -> {
+                BankAccountsView_Error()
+            }
+        }
+    }
+}
+
+@Composable
+fun BankAccountsView_Loading() {
+
+    Box(
+        modifier = Modifier
+            .height(120.dp)
+    ) {
+        Column(
+            modifier = Modifier.fillMaxSize(),
+            verticalArrangement = Arrangement.Center,
+            horizontalAlignment = Alignment.CenterHorizontally
+        ) {
+            Text(
+                text = "Checking Bank Accounts",
+                fontFamily = robotoFont,
+                fontWeight = FontWeight.Normal,
+                fontSize = 17.sp,
+                color = colorResource(id = R.color.primary_color)
+            )
+            CircularProgressIndicator(
+                modifier = Modifier
+                    .padding(top = 16.dp),
+                color = colorResource(id = R.color.primary_color)
+            )
+        }
+    }
+}
+
+@Composable
+fun BankAccountsView_Required(viewModel: BankAccountsViewModel?) {
```

Comment:
As mentioned in your previous PR. Can we put each of these Composable functions into their own file so the code is easier to read?